tests/integration/targets/win_pssession_configuration/tasks/main.yml

```diff
@@ -0,0 +1,18 @@
+# This file is part of Ansible
+
+# Copyright: (c) 2020, Brian Scholer <@briantist>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+---
+- block:
+    - ansible.windows.setup:
```

Comment:
Gathering facts can be an expensive operation and it looks like you just want to get the PowerShell version. I recommend just turning this to a `win_shell` task to get the `$PSVersion` to save on testing time.

Reply:
Does using `gather_subset` change that? By excluding everything else it seems to be running very fast for me.
No big deal to change it to `win_shell` just wondering if `gather_subset` is actually saving any execution time or just filtering out after the fact.

Reply:
Took a look at the `setup.ps1` code and it seems like it skips all the expensive stuff the way I'm calling it, which seems to agree with the performance I'm seeing (github isn't showing the `gather_subset` option when I view this as a comment on the issue conversation so you might have missed it).

If you'd prefer I'd change it to `win_shell` anyway let me know.

Reply:
If it works then let's just leave it be. `gather_subset` will definitely reduce the time it takes to run especially if you are only getting certain facts.